gae/room/protorpc_messages.py

```diff
@@ -991,6 +991,52 @@ class SiteCaptain(messages.Message):
   type = messages.StringField(4)
 
 
+############
+# StaffPosition #
+############
+def StaffPositionModelToMessage(mdl):
+  s = StaffPosition(
+    id=mdl.key.integer_id(),
+    name=mdl.position_name,
+    hourly_rate=mdl.GetHourlyRate(datetime.date.today()),
+    mileage_rate=mdl.GetMileageRate(datetime.date.today())
+  )
+  return s
+
+
+def StaffPositionMessageToModel(msg, mdl):
+  if not msg.name:
+    raise remote.ApplicationError('position name is required')
+  mdl.position_name = msg.name
+
+  today = datetime.date.today()
+  mdl.hourly_rate = mdl.GetHourlyRate(today)
+  mdl.mileage_rate = mdl.GetMileageRate(today)
+
+  if msg.hourly_rate and msg.hourly_rate != mdl.hourly_rate:
+    if not msg.hourly_start_date:
+      raise remote.ApplicationError('Hourly start date is required.')
+    mdl.hourly_rate_after_date += [msg.hourly_start_date + " " + str(msg.hourly_rate)]
+    mdl.hourly_rate = mdl.GetHourlyRate(today)
+
+  if msg.mileage_rate and msg.mileage_rate != mdl.mileage_rate:
+    if not msg.mileage_start_date:
+      raise remote.ApplicationError('Mileage start date is required.')
+    mdl.mileage_rate_after_date += [msg.mileage_start_date + " " + str(msg.mileage_rate)]
+    mdl.mileage_rate = mdl.GetMileageRate(today)
+
+  return mdl
+
+
+class StaffPosition(messages.Message):
+  id = messages.IntegerField(1)
+  name = messages.StringField(2)
+  hourly_rate = messages.FloatField(3)
+  hourly_start_date = messages.StringField(4)
+  mileage_rate = messages.FloatField(5)
+  mileage_start_date = messages.StringField(6)
+
```

Comment:
The way it's supposed to work is that a StaffPosition can have different rates for different date ranges.  

I think we want to model this as a repeated pair of dates and rates.
something like

class RateAfterDate(messages.Message):
  date = messages.StringField
  rate = messages.FloatField

class StaffPosition(messages.Message):
  id ...
  name ...
  hourly_rates = RateAfterDate(repeated=True)
  mileage_rates = RateAfterDate(repeated=True)


In the NDB model eventually we will want to model this with a Structured Property, but we can use a parsed string for now and do the conversion in a subsequent change.
https://cloud.google.com/appengine/docs/standard/python/ndb/entity-property-reference#structured